src/Modules/UngradedModules/SubquoModule.jl

```diff
@@ -804,14 +804,14 @@ function reduced_groebner_basis(M::SubquoModule, ord::ModuleOrdering = default_o
       M.groebner_basis[ord] = reduced_groebner_basis(M.sub, ord)
     else
       quo_gb = reduced_groebner_basis(M.quo, ord)
-      sub_union_gb_of_quo = SubModuleOfFreeModule(M.F, ModuleGens(vcat(M.sub.gens.O, quo_gb.O), M.F))
+      sub_union_gb_of_quo = SubModuleOfFreeModule(M.F, ModuleGens(vcat(oscar_generators(M.sub.gens), oscar_generators(quo_gb)), M.F))
       gb = reduced_groebner_basis(sub_union_gb_of_quo, ord)
       rel_gb_list = Vector{elem_type(ambient_free_module(M))}()
 
-      for i in 1:length(gb.O)
+      for i in 1:length(oscar_generators(gb))
```

Comment:
Same as my comment above on looping over oscar_generators and handling inside Singular elements.